File: source-code/Server/backend/src/main/java/com/iasdietideals24/backend/mapstruct/dto/NotificaDto.java
package com.iasdietideals24.backend.mapstruct.dto;

import com.iasdietideals24.backend.mapstruct.dto.shallows.AccountShallowDto;
import com.iasdietideals24.backend.mapstruct.dto.shallows.AstaShallowDto;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
public class NotificaDto {

    private Long idNotifica;

    private LocalDate dataInvio;

    private LocalTime oraInvio;

    private String messaggio;

    private AccountShallowDto mittenteShallow;

    private Set<AccountShallowDto> destinatariShallow;

    private AstaShallowDto astaAssociataShallow;
}

File: source-code/Client/frontendandroid/app/src/main/java/com/iasdietideals24/dietideals24/utilities/dto/NotificaDto.kt
package com.iasdietideals24.dietideals24.utilities.dto

import com.iasdietideals24.dietideals24.utilities.dto.shallows.AccountShallowDto
import com.iasdietideals24.dietideals24.utilities.dto.shallows.AstaShallowDto
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.time.LocalDate
import java.time.LocalTime

@Getter
@Setter
@NoArgsConstructor
class NotificaDto {
    private val idNotifica: Long? = null

    private val dataInvio: LocalDate? = null

    private val oraInvio: LocalTime? = null

    private val messaggio: String? = null

    private val mittenteShallow: AccountShallowDto? = null

    private val destinatariShallow: Set<AccountShallowDto>? = null

    private val astaAssociataShallow: AstaShallowDto? = null
}
